src/PSRule/Runtime/RunspaceContext.cs

```diff
@@ -614,6 +634,23 @@ public bool TrySelector(string name)
             return result;
         }
 
+        public bool MatchSuppressionGroup(out string suppressionGroupId)
+        {
+            suppressionGroupId = string.Empty;
+            foreach (var keyValuePair in Pipeline.SuppressionGroup)
+            {
+                var groupId = keyValuePair.Key;
+                var visitor = keyValuePair.Value;
+
+                if (visitor.Match(TargetObject.Value))
+                {
+                    suppressionGroupId = groupId;
+                    return true;
+                }
+            }
+            return false;
+        }
```

Comment:
@BernieWhite

I probably should find a more efficient way to search suppression groups. Currently this checks all suppression group visitors for each check. Would probably be much more efficient if we could skip suppression groups that are not relevant to the current rule block being checked or involve some caching of results. 

I thought about this but couldn't see a nice way to do it. Let me know what you think 🙂

Reply:
Maybe creating a dictionary where keys are rule names and values are the suppression group Ids? That could be a good start.

Reply:
I'd suggest extending `PSRule.Rules.SuppressionFilter`. We can probably rework it since it already has an index of rules.

Reply:
@BernieWhite Thanks. Yeah was thinking of doing that as well. I think that makes most sense.